MapboxDirections/MBVisualInstruction.swift

```diff
@@ -41,41 +41,53 @@ open class VisualInstruction: NSObject, NSSecureCoding {
      */
     @objc public let secondaryTextComponents: [VisualInstructionComponent]?
     
+    
+    /**
+     :nodoc:
+     Indicates what side of a bidirectional road the driver must be driving on. Also referred to as the rule of the road.
+     */
+    @objc public var drivingSide: DrivingSide
+    
     /**
      :nodoc:
      Initialize a `VisualInstruction` from a dictionary.
      */
-    @objc public convenience init(json: [String: Any]) {
+    @objc public convenience init(json: [String: Any], drivingSide: DrivingSide) {
         let distanceAlongStep = json["distanceAlongGeometry"] as! CLLocationDistance
         
+        
+        let maneuverType = ManeuverType(description: json["type"] as! String) ?? .none
+        let maneuverDirection = ManeuverDirection(description: json["modifier"] as! String)  ?? .none
+        
         let primaryTextComponent = json["primary"] as! JSONDictionary
         let primaryText = primaryTextComponent["text"] as! String
         let primaryTextComponents = (primaryTextComponent["components"] as! [JSONDictionary]).map {
-            VisualInstructionComponent(json: $0)
+            VisualInstructionComponent(json: $0, maneuverType: maneuverType, maneuverDirection: maneuverDirection)
         }
         
         var secondaryText: String?
         var secondaryTextComponents: [VisualInstructionComponent]?
         if let secondaryTextComponent = json["secondary"] as? JSONDictionary {
             secondaryText = secondaryTextComponent["text"] as? String
             secondaryTextComponents = (secondaryTextComponent["components"] as! [JSONDictionary]).map {
-                VisualInstructionComponent(json: $0)
+                VisualInstructionComponent(json: $0, maneuverType: maneuverType, maneuverDirection: maneuverDirection)
```

Comment:
@mapbox/navigation-ios  I'm not very into this, but it does address https://github.com/mapbox/MapboxDirections.swift/pull/239#discussion_r167655759. I think these visual classes should be refactored to be more inline with @frederoni proposed API. This should happen as tail work after this PR is merged.